tests/run/i18701.check

```diff
@@ -0,0 +1 @@
+public static final TB A.A$$anon$1$$_$_$$anonfun$1()
```

Comment:
I think this is not what's desired, the enclosing method should be `tb`. The `EnclosingMethod` attribute (aka `OUTERCLASS`) should represent the source level view.

There's a very long comment here: https://github.com/scala/scala3/blob/3.4.2/compiler/src/dotty/tools/backend/jvm/BTypes.scala#L341

There's a test suite for InnerClass / EnclosingMethod attributes at https://github.com/scala/scala/tree/2.13.x/test/files/jvm/innerClassAttribute which was not ported 1:1 to scala 3, but @prolativ  worked on the attributes in https://github.com/scala/scala3/pull/11589.